### PR TITLE
[Fixes #5680] Update groups to allow managers to promote regular memb…

### LINKED ISSUE #5680: Unable to add manager privileges to current group member
## Expected Behavior
I can **add** manager privileges to a current member of a group by the group UI if I am a manager of the group.

## Actual Behavior
I am unable to add manager privileges but have to remove the user from the group to add him back, or ask an admin to set him to group manager via the django-admin interface.

## Steps to Reproduce the Problem

  1. Add member to group without manager privileges
  2. Try to give him manager privileges like:

grafik

## Specifications

  - GeoNode version: 2.10.2.dev1581497314

### PR DESCRIPTION
**Pull request by travislbrundage**
…ers to manager without having to readd them

Fixes #5680 by adding a "promotion" button similar to "remove" button. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [ ] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/groups/templates/groups/group_members.html [hunk 1]
```diff
@@ -41,6 +41,13 @@ <h4>
                   {% csrf_token %}
                   <button type="submit" class="btn btn-danger btn-xs">{% trans 'Remove' %}</button>
                 </form>
+                {% if member.role == 'member' %}
+                  |
+                  <form style="display: inline;" method="POST" action="{% url "group_member_promote" group.slug member.user.username %}">
+                    {% csrf_token %}
+                    <button type="submit" class="btn btn-success btn-xs">{% trans 'Promote' %}</button>
```
Comment (gannebamm) [inline geonode/groups/templates/groups/group_members.html]: If a new translation string is added, should we also update the django.po files? I have done a __very quick__ search for `Promote` as string and have not found it in EN locale.

**Comment by afabiani** [comment]
@gannebamm you make some comments on the translations but then approved the PR.

I guess this is not ready yet to be merged. Moreover, since this one seems to be new functionality and not a bug fix, I would add:

 - a test case
 - update the documentation (if it does exist) or at least open an issue on the geonode-documentation to add the missing parts and refer them here

**Comment by gannebamm** [comment]
@afabiani yes, you are right. Shouldn´t have approved it, yet.

**Comment by travislbrundage** [comment]
Will add this today, thank you for your feedback

**Comment by travislbrundage** [comment]
So I updated this a bit to include both "promote" and "demote", since I figured it only truly satisfied the issue if you could go back too.

I am not sure how to update the documentation with appropriate, updated pictures, but I've given it a quick update here: https://github.com/GeoNode/documentation/pull/16

And made an issue here: https://github.com/GeoNode/documentation/issues/15

**Pull request merged**